test/commands/auth/device/login.test.ts

```diff
@@ -82,151 +83,103 @@ describe('auth:device:login', async () => {
-      expect(response.result.username).to.equal(testData.username);
-    });
-
-  test
-    .do(async () => {
-      await prepareStubs();
-      $$.SANDBOX.stub(Global, 'getEnvironmentMode').returns(Mode.DEMO);
-      uxStub = stubMethod($$.SANDBOX, UX.prototype, 'prompt').resolves('yes');
-    })
-    .stdout()
-    .command(['auth:device:login', '--json'])
-    .it('should prompt for when in demo mode (SFDX_ENV=demo)', (ctx) => {
-      const [action, response] = parseJsonResponse(ctx.stdout);
-      expect(uxStub.callCount).to.equal(1);
-      expect(action).to.deep.equal(mockAction);
-      expect(response.status).to.equal(0);
-      expect(response.result.username).to.equal(testData.username);
-    });
-
-  test
-    .do(async () => {
-      await prepareStubs();
-      $$.SANDBOX.stub(Global, 'getEnvironmentMode').returns(Mode.DEMO);
-      uxStub = stubMethod($$.SANDBOX, UX.prototype, 'prompt').resolves('NO');
-    })
-    .finally(() => {
-      delete process.env['SFDX_ENV'];
-    })
-    .stdout()
-    .command(['auth:device:login', '--json'])
-    .it('should exit early when prompt is answered NO', (ctx) => {
-      const response = parseJson<AuthFields>(ctx.stdout);
-      expect(uxStub.callCount).to.equal(1);
-      expect(response.status).to.equal(0);
-      expect(response.result).to.deep.equal({});
-    });
+  it('should return auth fields', async () => {
+    await prepareStubs();
+    const login = new Login(['--json'], {} as Config);
+    const response = await login.run();
+    expect(response.username).to.equal(testData.username);
+  });
+
+  it('should return auth fields with instance url', async () => {
+    await prepareStubs();
+    const login = new Login(['-r', 'https://login.salesforce.com', '--json'], {} as Config);
+    const response = await login.run();
+    expect(response.username).to.equal(testData.username);
+  });
+
+  it('should set alias when -a is provided', async () => {
+    await prepareStubs();
+    const login = new Login(['-a', 'MyAlias', '--json'], {} as Config);
+    const response = await login.run();
+    expect(response.username).to.equal(testData.username);
+    expect(authInfoStub.handleAliasAndDefaultSettings.callCount).to.equal(1);
+  });
+
+  it('should set target-org when -s is provided', async () => {
+    await prepareStubs();
+    const login = new Login(['-s', '--json'], {} as Config);
+    const response = await login.run();
+    expect(response.username).to.equal(testData.username);
+    expect(authInfoStub.handleAliasAndDefaultSettings.callCount).to.equal(1);
+  });
+
+  it('should set target-dev-hub when -d is provided', async () => {
+    await prepareStubs();
+    const login = new Login(['-d', '--json'], {} as Config);
+    const response = await login.run();
+    expect(response.username).to.equal(testData.username);
+    expect(authInfoStub.handleAliasAndDefaultSettings.callCount).to.equal(1);
+  });
+
+  // not sure how to handle this
```

Comment:
stub the ux method you need
run the command from its static `run` method.
assert the arguments received by the stub